docs-user/bunny.md

```diff
@@ -0,0 +1,272 @@
+
+The thread list also nicely shows the message passing between the content process' main thread and the worker thread. The content process posts a message, and then effectively waits for a response before it does anything. This is a fairly common pattern to see in multi-threaded code.
+
+### Problems in the content process' main thread
+
+The Flame Graph provides a nice view into a summary of where time is spent. The X axis represents the percentage of time spent with that function in that stack for all visible stacks. In a previous step the idle time was already hidden from the analysis.
+
+The stacks are rather deep here, so a nice first step is to focus on just the subtree that is interesting. Visually, `nsThread::ProcessNextEvent` is the last function most common to the tree. Right click and focus on that subtree.
+
+![A screenshot of the context menu on the flame graph for focus subtree.](./images/bunny-analysis/focus-subtree.png)
+
+Two functions really stand out as taking a lot of time. `JSStructuredCloneReader::read` takes almost 30% of the time. It is a C++ function, and is called when the iframe receives a message from the worker. It safely reads a clone of the data, and provides it to the iframe's JavaScript code.
+
+The bigger culprit is `drawRects`, which takes 60% of the time. This is the function that calls out the `CanvasRenderingContext2D` API to actually draw to the screen. There are two functions that are being called from drawRects. These are `set CanvasRenderingContext2D.fillStyle` and `CanvasRenderingContext2D.fillRect`.
+
+[https://perfht.ml/2Ios9PH](https://perfht.ml/2Ios9PH)
+
+![Screenshot of the flame graph of the content process.](./images/bunny-analysis/flame-graph-content.png)
+
+### Problems in the worker
+
+Looking at the worker thread, first focus on the subtree of `(root scope) https://glittr-sandbox:4444/squares/worker.js`, as this contains the relevant code for the analysis.
+
+[https://perfht.ml/2Iu4mh4](https://perfht.ml/2Iu4mh4)
+
+![Screenshot of the flame graph of the worker process.](./images/bunny-analysis/flame-graph-worker.png)
+
+There are two main functions that really stand out as taking up a large amount of time. The first is the `drawLineFromPoints`. This happens to be the user's code that is evaluated. Most of it we don't have control over. the `rect` function shows up, but it is a small part of the total time. `DedicatedWorkerGlobalScope.postMessage` and `JSStructuredCloneWriter::write` show up as taking most of the time. This is the part of the code where the worker is posting a message back to the iframe's JavaScript.
+
+## Hypothesis
+
+Based on this baseline report, the two areas that seem to be problematic are the `fillStyle` and and the structured clone behavior with posting messages. Fixing these will significantly speed up the frame rate.
+
+## Fixing `set fillStyle`
```

Comment:
nit: the style for `set fillStyle` in a title does a too small font-size -- likely because we use `px` instead of `em` to define the monospace font size value :-)

![image](https://user-images.githubusercontent.com/454175/40552029-e4c8f7fe-603e-11e8-8772-83ac00035594.png)

(same below)